docs/developer-feature-tutorial.md

```diff
@@ -99,74 +132,104 @@ Now, run zulip terminal and open web app from a different account. Start composi
-                    ' is typing...'
-                ])
-```
-changes the footer text, and this
+These are done by the highlighted lines accordingly in `set_footer_text`
 ```python
-self.controller.update_screen()
+@asynch
+def set_footer_text(self, text_list: Optional[List[Any]]=None, duration: Optional[float]=None) -> None:
+    if text_list is None:
+        text = self.get_random_help()
+    else:
+        text = text_list
+>>> self._w.footer.set_text(text)
+>>> self.controller.update_screen()
+    if duration is not None:
+        assert duration > 0
+        time.sleep(duration)
+        self.set_footer_text()
 ```
-updates the screen to display the changes. This fully implements the typing feature.
 
-### Writing tests
+## Writing tests
+
+We use `pytest` to write all our tests. We have to write new tests, modify current ones or delete old ones whenever needed.
 
-Now, we update the tests for `register_initial_desired_events` by adding `typing`
+For this feature, we update the tests for `register_initial_desired_events` by adding `typing`
 to the event types.
-```diff
+```python
 # tests/core/test_core.py
-
-        event_types = [
-            'message',
-            'update_message',
-            'reaction',
-+           'typing',
-        ]
-        controller.client.register.assert_called_once_with(
-                                   event_types=event_types,
-                                   apply_markdown=True)
+  event_types = [
+      'message',
+      'update_message',
+      'reaction',
+      'subscription',
+>>>   'typing',
+      'update_message_flags',
+      'update_display_settings',
+  ]
 ```
 
-`test_handle_typing_event` in `test_ui.py` implements testing for `handle_typing_event`. Please read it to understand how to write tests for a new function in zulip terminal.
+and we create a new function `test_handle_typing_event` in `test_model.py` which implements testing for `handle_typing_event`. 
+
+Please read it to understand how to write tests for a new function in zulip terminal.
+
+How to run these tests? Easy:
+* To run all tests just type `pytest` in your terminal.
+* To run all linting tests type `./tools/lint-all` in your terminal.
+
+## Send Pull Request
+Remember to delete the modification in `poll_for_events()` and delete the file `type_test.json` before you are ready to commit.
+
+* Use `git status` to check your branch and files changed.
+* `git diff` to verify changes.
+* `git add` to stage changes.
+* `git commit` according to the **commit guidelines**.
+* `git rebase -i` Rebase in interactive mode to cleanup you commit. _(Lookup squashing)_
+* Finally `git push origin feature-typing` to push to your fork in Github.
+
+Now you can go to Github and create a **Pull Request** from the **feature branch** to the main Zulip Terminal repository to be reviewed.
+
+Do send in **[WIP]** PR's or Work In Progress PR's regularly so that we can keep track of your work and give advise when needed.
```

Comment:
"Give advise" can be changed to either "give advice" or just "advise". See https://www.grammarly.com/blog/advise-advice/.